lib/service/connect.go

```diff
@@ -76,6 +85,21 @@ func (process *TeleportProcess) reconnectToAuthService(role types.SystemRole) (*
 	}
 }
 
+func (process *TeleportProcess) compareVersions(resp *proto.PingResponse) error {
+	serverVersion, err := semver.NewVersion(resp.ServerVersion)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	teleportVersion, err := semver.NewVersion(teleport.Version)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	if math.Abs(float64(serverVersion.Major-teleportVersion.Major)) > 1 {
+		process.log.Warnf("Only versions %s and %s are supported, auth server is %s.", teleportVersion.Major, teleportVersion.Major-1, serverVersion.Major)
```

Comment:
Typoed my suggestion earlier, these are numbers

```suggestion
		process.log.Warnf("Only versions %d and %d are supported, auth server is %d.", teleportVersion.Major, teleportVersion.Major-1, serverVersion.Major)
```